### PR TITLE
Add configuration via system console

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
I wanted to make this plugin configurable via the System Console, since doing it with mmctl was causing me issues and I wanted to update it often. But, because the plugin manifest settings schema only offers strings and other primitive types as options, I needed the configuration for `WelcomeMessages` to support being either a string or a JSON array. Hence, the custom `UnmarshalJSON` for `Configuration`.

It's not an ideal solution but it works decently well. The other method I considered taking was using the `custom` settings schema type but then I'd have to re-implement the text field component and add a web app component to this plugin, which I didn't want to do. Thoughts?

### CODE REVIEW

**Review by levb — APPROVED** [review]

File: server/manifest.go
Comment (levb) [inline server/manifest.go:10]: @jwilander we should probably leave the version out of this PR, and then once merged kick off the "next release" playbook

File: README.md
Comment (jfrerich) [inline README.md:21]:
```suggestion
1. Go to **System Console > PLUGINS > Plugin Management** and click **Enable** to enable the Welcome Bot plugin.
```

`Plugins` is the parent section name for `Plugin Management`.  This is a small change and I'm 0/5 whether it should be included, but think it is more clear on how to find the `Pluigin Management`.

Comment (jfrerich) [reply README.md:21]: Nice improvement, @jwilander.  I only have one suggestion with regard to the README.

**Review by jfrerich — APPROVED** [review]
Approving with one nit suggestion.

**Review by catalintomai — APPROVED** [review]

**Comment by hanzei** [comment]
@AayushChaudhary0001 Would you be open to reviewing this PR?

**Comment by mickmister** [comment]
Note that there is a similar PR being finished up that implements a custom admin component for this https://github.com/Brightscout/mattermost-plugin-welcomebot/pull/15. I think this PR can/should be closed in favor of the custom component implementation. Are you okay with this @jwilander?

**Comment by jwilander** [comment]
Yes, let's close it favour of that.

**Comment by Kshitij-Katiyar** [comment]
Opening this PR due to comments https://github.com/mattermost/mattermost-plugin-welcomebot/issues/108#issuecomment-2005154745 and https://github.com/mattermost/mattermost-plugin-welcomebot/issues/108#issuecomment-2003521757.

**Review by AayushChaudhary0001 — COMMENTED** [review]
@Kshitij-Katiyar While testing this PR, it was found that this was not working for MM cloud server. Any new user added to a team was not getting any welcome message nor any action button. Please verify it once.

**Comment by Kshitij-Katiyar** [comment]
Closing the PR because of the PR https://github.com/mattermost/mattermost-plugin-welcomebot/pull/126

**Pull request closed without merging**
